src/main/java/org/jenkinsci/plugins/xunit/pipeline/XUnitResultsStepExecution.java

```diff
@@ -0,0 +1,63 @@
+package org.jenkinsci.plugins.xunit.pipeline;
+
+import hudson.AbortException;
+import hudson.FilePath;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.junit.TestResultAction;
+import hudson.tasks.junit.TestResultSummary;
+import hudson.tasks.junit.pipeline.JUnitResultsStepExecution;
+import org.jenkinsci.lib.dtkit.type.TestType;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.SynchronousStepExecution;
+import org.jenkinsci.plugins.xunit.ExtraConfiguration;
+import org.jenkinsci.plugins.xunit.XUnitProcessor;
+import org.jenkinsci.plugins.xunit.threshold.XUnitThreshold;
+
+import javax.annotation.Nonnull;
+
+public class XUnitResultsStepExecution extends SynchronousStepExecution<TestResultSummary> {
```

Comment:
I found some problems with `SynchronousNonBlockingStepExecution`, if I remember correctly - the step runs could stomp on each other and cause problems. What I want is a guarantee that no two instances of `xunit` can be running at the same time in the same Pipeline.